semgrep/semgrep/core_runner.py

```diff
@@ -270,21 +269,41 @@ def _run_rule(
                     debug_print(core_run.stderr.decode("utf-8", "replace"))
 
                     if core_run.returncode != 0:
-                        try:
-                            # see if semgrep output a JSON error that we can decode
-                            semgrep_output = core_run.stdout.decode("utf-8", "replace")
-                            output_json = json.loads(semgrep_output)
-                            if "error" in output_json:
-                                self._semgrep_error_json_to_message_then_exit(
-                                    output_json
-                                )
-                            else:
-                                raise SemgrepError(
-                                    f"unexpected non-json output while invoking semgrep-core:\n{PLEASE_FILE_ISSUE_TEXT}"
-                                )
-                        except Exception as e:
```

Comment:
I removed this try-catch-exception because it's going to error anyway and it swallows a lot of coding bugs in a hard-to-debug way. We'll still get these errors at the top level for debugging user issues.